src/prism.c

```diff
@@ -13549,8 +13607,13 @@ parse_pattern_primitive(pm_parser_t *parser, pm_diagnostic_id_t diag_id) {
                     parser_lex(parser);
                     pm_node_t *variable = (pm_node_t *) parse_variable(parser);
                     if (variable == NULL) {
-                        PM_PARSER_ERR_TOKEN_FORMAT(parser, parser->previous, PM_ERR_NO_LOCAL_VARIABLE, (int) (parser->previous.end - parser->previous.start), parser->previous.start);
-                        variable = (pm_node_t *) pm_local_variable_read_node_create(parser, &parser->previous, 0);
+                        if (pm_token_is_it(parser->previous.start, parser->previous.end)) {
```

Comment:
We need to gate this on version as well.

```suggestion
                        if (parser->version != PM_OPTIONS_VERSION_CRUBY_3_3_0 && pm_token_is_it(parser->previous.start, parser->previous.end)) {
```